Intermittent Search test failures

This fails pretty frequently, but not 100% of the time. And if you re-trigger a build on Travis it will usually pass after the failure.

So... it would be good to figure out why this happens (and why it's a recent thing).

```
FAIL: main - Search (non-auth) - Search works (3496ms)

AssertionError: The search results show up on the page.: expected 'Issue 280: w3.org - desktop site instead of mobile site' to include 'vladvlad'

  at Function.assert.include  <node_modules/intern/node_modules/chai/chai.js:2977:45>

  at CompatCommand.<anonymous>  <tests/functional/search-non-auth.js:92:18>

  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:509:31>

  at CompatCommand.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:526:11>

  at <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:235:37>

  at runCallbacks  <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:11:18>

  at <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:188:17>

  at Object.run [as _onImmediate]  <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:131:29>

  at processImmediate [as _immediateCallback]  <timers.js:354:15>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:525:10>

  at Test.registerSuite.Search works [as test]  <tests/functional/search-non-auth.js:91:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.end  <node_modules/intern/node_modules/leadfoot/Command.js:475:10>

  at Test.registerSuite.Search works [as test]  <tests/functional/search-non-auth.js:94:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

127.0.0.1 - - [30/Dec/2015 02:19:03] "GET / HTTP/1.1" 200 -

Expected fixture file: /home/travis/build/webcompat/webcompat.com/tests/fixtures/api/issues/labels.6665c4820456e6a813dcbbdddd690ffc.json

127.0.0.1 - - [30/Dec/2015 02:19:03] "GET /api/issues/labels?per_page=100 HTTP/1.1" 200 -

127.0.0.1 - - [30/Dec/2015 02:19:03] "GET /api/issues/category/new HTTP/1.1" 200 -

127.0.0.1 - - [30/Dec/2015 02:19:04] "GET /issues?q=vladvlad HTTP/1.1" 200 -

Expected fixture file: /home/travis/build/webcompat/webcompat.com/tests/fixtures/api/issues/labels.6665c4820456e6a813dcbbdddd690ffc.json

127.0.0.1 - - [30/Dec/2015 02:19:04] "GET /api/issues/labels?per_page=100 HTTP/1.1" 200 -

FAIL: main - Search (non-auth) - Search from the homepage (14080ms)

NoSuchElement: [POST http://localhost:4444/wd/hub/session/a82eeaae-3abd-4880-a73f-bab8a12c0b58/element / {"using":"css selector","value":".wc-IssueItem:nth-of-type(1) a"}] Unable to locate element: {"method":"css selector","selector":".wc-IssueItem:nth-of-type(1) a"}

For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html

Build info: version: '2.48.0', revision: 'b7b081a', time: '2015-10-07 15:48:56'

System info: host: 'testing-worker-linux-docker-8399f37a-3194-linux-5', ip: '172.17.1.255', os.name: 'Linux', os.arch: 'amd64', os.version: '3.13.0-40-generic', java.version: '1.7.0_76'

Driver info: driver.version: unknown

  at Server._post  <node_modules/intern/node_modules/leadfoot/Server.js:68:9>

  at ProxiedSession._post  <node_modules/intern/node_modules/leadfoot/Session.js:57:30>

  at ProxiedSession.Session.find  <node_modules/intern/node_modules/leadfoot/Session.js:1053:15>

  at CompatCommand.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:41:36>

  at <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:235:37>

  at runCallbacks  <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:11:18>

  at <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:188:17>

  at Object.run [as _onImmediate]  <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:131:29>

  at processImmediate [as _immediateCallback]  <timers.js:354:15>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.find  <node_modules/intern/node_modules/leadfoot/Command.js:22:10>

  at <CompatCommand.prototype.(anonymous function) [as findByCssSelector] (/home/travis/build/webcompat/webcompat.com/node_modules/intern/node_modules/leadfoot/lib/strategies.js:24:16)>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:108:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at <CompatCommand.target.(anonymous function) [as getVisibleText] (/home/travis/build/webcompat/webcompat.com/node_modules/intern/node_modules/leadfoot/Command.js:656:11)>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:108:62>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:525:10>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:109:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.end  <node_modules/intern/node_modules/leadfoot/Command.js:475:10>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:112:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at <CompatCommand.target.(anonymous function) [as getCurrentUrl] (/home/travis/build/webcompat/webcompat.com/node_modules/intern/node_modules/leadfoot/Command.js:592:11)>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:113:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:525:10>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:114:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.end  <node_modules/intern/node_modules/leadfoot/Command.js:475:10>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:117:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

2/6 tests failed
```

--- a/tests/fixtures/api/issues/search.658ca355332f37516ee5154982b283c5.json
+++ b/tests/fixtures/api/issues/search.658ca355332f37516ee5154982b283c5.json
@@ -1,0 +1,54 @@
+{
+  "_fixture": "true",
+  "total_count": 1,
+  "incomplete_results": false,
+  "items": [
+    {
+      "url": "https://api.github.com/repos/webcompat/webcompat-tests/issues/114",
+      "repository_url": "https://api.github.com/repos/webcompat/webcompat-tests",
+      "labels_url": "https://api.github.com/repos/webcompat/webcompat-tests/issues/114/labels{/name}",
+      "comments_url": "https://api.github.com/repos/webcompat/webcompat-tests/issues/114/comments",
+      "events_url": "https://api.github.com/repos/webcompat/webcompat-tests/issues/114/events",
+      "html_url": "https://github.com/webcompat/webcompat-tests/issues/114",
+      "id": 37931436,
+      "number": 114,
+      "title": "vladvlad",
+      "user": {
+        "login": "testerbuddy2",
+        "id": 8173975,
+        "avatar_url": "https://avatars.githubusercontent.com/u/8173975?v=3",
+        "gravatar_id": "",
+        "url": "https://api.github.com/users/testerbuddy2",
+        "html_url": "https://github.com/testerbuddy2",
+        "followers_url": "https://api.github.com/users/testerbuddy2/followers",
+        "following_url": "https://api.github.com/users/testerbuddy2/following{/other_user}",
+        "gists_url": "https://api.github.com/users/testerbuddy2/gists{/gist_id}",
+        "starred_url": "https://api.github.com/users/testerbuddy2/starred{/owner}{/repo}",
+        "subscriptions_url": "https://api.github.com/users/testerbuddy2/subscriptions",
+        "organizations_url": "https://api.github.com/users/testerbuddy2/orgs",
+        "repos_url": "https://api.github.com/users/testerbuddy2/repos",
+        "events_url": "https://api.github.com/users/testerbuddy2/events{/privacy}",
+        "received_events_url": "https://api.github.com/users/testerbuddy2/received_events",
+        "type": "User",
+        "site_admin": false
+      },
+      "labels": [
+        {
+          "url": "https://api.github.com/repos/webcompat/webcompat-tests/labels/closed-duplicate",
+          "name": "closed-duplicate",
+          "color": "cccccc"
+        }
+      ],
+      "state": "open",
+      "locked": true,
+      "assignee": null,
+      "milestone": null,
+      "comments": 9,
+      "created_at": "2014-07-15T22:14:21Z",
+      "updated_at": "2014-07-21T14:56:05Z",
+      "closed_at": null,
+      "body": "<!-- @browser: Chrome 35.0.1916 -->\r\n\r\n**URL**: test.com\r\n**Browser / Version**: Chrome 35.0.1916\r\n**Operating System**: Mac OS X 10.9.2\r\n**Problem type**: Unknown\r\n**Site owner**: Unknown\r\n\r\n**Steps to Reproduce**\r\n1) Navigate to: test.com\r\n2) …\r\n\r\nExpected Behavior:\r\nActual Behavior:\r\n\r\n```css\r\nfloat: left\r\n```\r\n",
+      "score": 32.393833
+    }
+  ]
+}

--- a/tests/functional/search-auth.js
+++ b/tests/functional/search-auth.js
@@ -71,6 +71,24 @@ define([
         .end();
     },
 
+    'Search works': function() {
+      return this.remote
+        .setFindTimeout(intern.config.wc.pageLoadTimeout)
+        .get(require.toUrl(url('/issues')))
+        .findByCssSelector('.js-SearchForm input')
+        .type('vladvlad')
+        .end()
+        .findByCssSelector('.js-SearchForm button').click()
+        .end()
+        // this is lame, but we gotta wait on search results.
+        .sleep(3000)
+        .findByCssSelector('.wc-IssueList:nth-of-type(1) a').getVisibleText()
+        .then(function(text) {
+          assert.include(text, 'vladvlad', 'The search results show up on the page.');
+        })
+        .end();
+    },
+
     'Search from the homepage': function() {
       return this.remote
         .setFindTimeout(intern.config.wc.pageLoadTimeout)

--- a/tests/functional/search-non-auth.js
+++ b/tests/functional/search-non-auth.js
@@ -8,7 +8,7 @@ define([
   'intern/chai!assert',
   'require',
   'intern/dojo/node!leadfoot/keys'
-], function(intern, registerSuite, assert, require, keys) {
+], function(intern, registerSuite, assert, require) {
   'use strict';
 
   var url = function(path) {
@@ -72,47 +72,6 @@ define([
         .findByCssSelector('.js-SearchForm').isDisplayed()
         .then(function(isDisplayed) {
           assert.equal(isDisplayed, true, 'Search input is visible for non-authed users.');
-        })
-        .end();
-    },
-
-    'Search works': function() {
-      return this.remote
-        .setFindTimeout(intern.config.wc.pageLoadTimeout)
-        .get(require.toUrl(url('/issues')))
-        .findByCssSelector('.js-SearchForm input')
-        .type('vladvlad')
-        .end()
-        .findByCssSelector('.js-SearchForm button').click()
-        .end()
-        // this is lame, but we gotta wait on search results.
-        .sleep(3000)
-        .findByCssSelector('.wc-IssueList:nth-of-type(1) a').getVisibleText()
-        .then(function(text) {
-          assert.include(text, 'vladvlad', 'The search results show up on the page.');
-        })
-        .end();
-    },
-
-    'Search from the homepage': function() {
-      return this.remote
-        .setFindTimeout(intern.config.wc.pageLoadTimeout)
-        .get(require.toUrl(url('/')))
-        .findByCssSelector('.js-SearchBarOpen').click()
-        .end()
-        .findByCssSelector('.js-SearchBar input').click()
-        .type('vladvlad')
-        .type(keys.ENTER)
-        .end()
-        .sleep(3000)
-        .findByCssSelector('.wc-IssueList:nth-of-type(1) a').getVisibleText()
-        .then(function(text) {
-          assert.include(text, 'vladvlad', 'The search query results show up on the page.');
-        })
-        .end()
-        .getCurrentUrl()
-        .then(function(currUrl) {
-          assert.include(currUrl, 'page=1', 'Default params got merged.');
         })
         .end();
     }

